src/main/java/LocalKeyClientEncryption.java

```diff
@@ -0,0 +1,175 @@
+
+        // Creating a blob client
+        BlobClient blobClient = blobContainerClient.getBlobClient(exampleBlobName + exampleBlobSuffix);
+        // Creating a local key and setting encryptedKeyClient
+        this.createLocalKey();
+        this.setLocalKeyClient(this.key, blobClient);
+
+        // Uploading example blob with client-side encryption
+        String fileName = exampleBlobName + exampleBlobSuffix;
+        this.encryptedBlobClient.uploadFromFile(".\\src\\main\\java\\" + fileName);
+    }
+
+    /**
+     * Downloads client-side encrypted blob, decrypts with local key, then reuploads with server-side encryption
+     */
+    public void decryptReupload(String storageAccount, String sharedKeyCred, String containerName, String blobName,
+                                String blobSuffix, AsyncKeyEncryptionKey key, CustomerProvidedKey serverKey){
+        String storageAccountUrl = "https://" + storageAccount + ".blob.core.windows.net";
+
+        // Creating blob client to download blob
+        BlobClient blobClient = new BlobClientBuilder()
+                .endpoint(storageAccountUrl)
+                .credential(new StorageSharedKeyCredential(storageAccount, sharedKeyCred))
+                .containerName(containerName)
+                .blobName(blobName + blobSuffix)
+                .buildClient();
+
+        // Set blob encryption client
+        if (key == null) {
+            key = this.key;
+        }
+        this.setLocalKeyClient(key, blobClient);
+
+        // Downloading encrypted blob, blob is decrypted upon download
+        String fileName = blobName + "Decrypted" + blobSuffix;
+        this.encryptedBlobClient.downloadToFile(".\\src\\main\\java\\" + fileName);
+
+        // Creating blob client for decryption and reuploading
+        BlobClientBuilder blobClientBuilder = new BlobClientBuilder()
+                .endpoint(storageAccountUrl)
+                .credential(new StorageSharedKeyCredential(storageAccount, sharedKeyCred))
+                .containerName(containerName)
+                .blobName(fileName);
+
+        // Checking to see if local key is provided to use customer-managed key for server encryption
+        if (serverKey != null){
+            String CPKfileName = blobName + "DecryptedCPK" + blobSuffix;
+            blobClientBuilder = blobClientBuilder
+                    .customerProvidedKey(serverKey)
+                    .blobName(CPKfileName);
+        }
+        BlobClient blobClientDecrypted = blobClientBuilder.buildClient();
+
+        // Uploading file to server
+        blobClientDecrypted.uploadFromFile(".\\src\\main\\java\\" + fileName);
+
+        // Cleaning up by deleting local save of encrypted blob
+        File localFile = new File(".\\src\\main\\java\\" + fileName);
+        localFile.delete();
+
+    }
+
+    /**
+     * Requires setting and naming the following environmental variables (must be in system environmental variables):
+     * sasToken (for storage account)
```

Comment:
nit. docs still mention sasToken. I think some of this setup stuff can also go in a README file - just explaining how a user should set themselves up to run this program successfully